pluginlib/package.xml

```diff
@@ -22,7 +22,8 @@
 
   <build_depend>cmake_modules</build_depend>
 
-  <depend>boost</depend>
+  <depend>libboost-filesystem-dev</depend>
+  <depend>libboost-system-dev</depend>
```

Comment:
Wouldn't be better having those inside a <build_depend> instead?

As state here: https://docs.ros.org/indigo/api/catkin/html/howto/format2/system_library_dependencies.html

> It is not recommended for system dependencies, because it forces your package’s binary installation to depend on the “development” package, which is not generally necessary or desirable

Reply:
Thanks for the feedback

This PR has been mostly automated as a first step in an effort to get rid of `libboost-all-dev` in all of `ros_core`, and did not audit each package closely.
Currently pluginlib is using `<depend>boost</depend>` that pulls `libboost-all-dev` / ~800MB of development packages.
This change is an incremental improvement as after this change it will pull "only" 200.

---

Looking closer at this package:
Pluginlib is a header-only library, so it *has to* export the dependencies downstream users need install to be able to link against pluginlib. So it cannot be only a `build_depend`.

It can however be changed to be a simple `build_export_depend`. The same would apply to all `<depend>` listed in this package.